### PR TITLE
Added Top App Bar Demo to Material section

### LINKED ISSUE #189: Add Top App Bar Demo to Material section

### PR DESCRIPTION
**Pull request by Alabhya268**
Gallery side - Closes #189 

image
image

### CODE REVIEW

**Comment by Alabhya268** [comment]
@rami-a Please have a look at this PR.

File: lib/data/demos.dart [hunk 1]
```diff
@@ -201,6 +202,22 @@ List<GalleryDemo> materialDemos(GalleryLocalizations localizations) {
       ],
       category: GalleryDemoCategory.material,
     ),
+    GalleryDemo(
```
Comment (rami-a) [inline lib/data/demos.dart:205]: This demo should be ordered alphabetically based on its title. Also I think the title should be updated to just `App bar` to match the widget itself. And then put it alphabetically.

File: lib/data/icons.dart [hunk 2]
```diff
@@ -171,4 +171,8 @@ class GalleryIcons {
     60080,
     fontFamily: 'MaterialIcons',
   );
+  static const IconData topAppbar = IconData(
+    60134,
```
Comment (rami-a) [inline lib/data/icons.dart:175]: Nice!

File: lib/demos/material/top_app_bar_demo.dart [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+class TopAppBarDemo extends StatelessWidget {
+  const TopAppBarDemo({Key key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    var localization = GalleryLocalizations.of(context);
+    return Scaffold(
+      appBar: AppBar(
+        leading: const Icon(
+          Icons.tag_faces,
```
Comment (rami-a) [inline lib/demos/material/top_app_bar_demo.dart:19]: Each IconButton needs a semanticLabel for accessibility. I think you should use the same icons in this demo as the `BottomAppBar` demo so that we do not need additional translations.

File: lib/demos/material/top_app_bar_demo.dart [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        ),
+        title: Text(localization.demoTopAppBarTitle),
+        actions: <Widget>[
```
Comment (rami-a) [inline lib/demos/material/top_app_bar_demo.dart:22]: Remove `<Widget>`

File: lib/demos/material/top_app_bar_demo.dart [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+class TopAppBarDemo extends StatelessWidget {
+  const TopAppBarDemo({Key key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    var localization = GalleryLocalizations.of(context);
+    return Scaffold(
+      appBar: AppBar(
+        leading: const Icon(
```
Comment (rami-a) [inline lib/demos/material/top_app_bar_demo.dart:18]: Make this an IconButton

File: lib/demos/material/top_app_bar_demo.dart [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+              Icons.directions_bus,
+            ),
+            onPressed: () {},
+          ),
+          PopupMenuButton<Text>(
+            itemBuilder: (context) {
+              return [
+                PopupMenuItem(
+                  child: Text(
+                    localization.demoTopAppBarMenuItemText1,
```
Comment (rami-a) [inline lib/demos/material/top_app_bar_demo.dart:40]: Try to reuse more generic already translated strings from other demos, to reduce the translation work for this project.

File: lib/demos/material/top_app_bar_demo.dart [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+                  ),
+                ),
+                PopupMenuItem(
+                  child: Text(
+                    localization.demoTopAppBarMenuItemText2,
```
Comment (rami-a) [inline lib/demos/material/top_app_bar_demo.dart:45]: Same here for using more generic already translated strings, look at the list demo for example

File: lib/l10n/intl_en.arb [hunk 8]
```diff
@@ -593,6 +593,30 @@
   "@settingsAttribution": {
     "description": "Title for attribution (TOASTER is a proper name and should remain in English)."
   },
+  "demoTopAppBarTitle": "Top app bar",
```
Comment (rami-a) [inline lib/l10n/intl_en.arb:596]: Make this just `App bar` like mentioned above

File: lib/l10n/intl_en.arb [hunk 9]
```diff
@@ -593,6 +593,30 @@
...
+  "@demoTopAppBarTitle": {
+    "description": "Title for the material Top app bar component demo."
+  },
+  "demoTopAppBarSubtitle": "A material design app bar.",
```
Comment (rami-a) [inline lib/l10n/intl_en.arb:600]: Make this: `Displays information and actions relating to the current screen`

**Review by rami-a — COMMENTED** [review]
Awesome! Just a few suggestions and changes

**Comment by Alabhya268** [comment]
@rami-a  I've made necessary changes .
Please have a look.

image
image

File: lib/demos/material/app_bar_demo.dart [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+class AppBarDemo extends StatelessWidget {
+  const AppBarDemo({Key key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    var localization = GalleryLocalizations.of(context);
+    return Scaffold(
+      appBar: AppBar(
+        leading: IconButton(
+          icon: const Icon(Icons.menu),
```
Comment (rami-a) [inline lib/demos/material/app_bar_demo.dart:19]: Please set the tooltip params for each IconButton, just like in https://github.com/flutter/gallery/blob/master/lib/demos/material/bottom_app_bar_demo.dart#L142

**Review by rami-a — APPROVED** [review]
Almost good to merge, just a few small changes

**Comment by rami-a** [comment]
I think you'll have to update some of the golden tests since that check is failing. See instructions at https://github.com/flutter/gallery/tree/master/golden_test

**Comment by Alabhya268** [comment]
> I think you'll have to update some of the golden tests since that check is failing. See instructions at https://github.com/flutter/gallery/tree/master/golden_test

@rami-a I checked, the golden test is failing on master branch too .

**Comment by rami-a** [comment]
https://github.com/flutter/gallery/pull/375 was just merged and updated some golden tests. Could you please update your branch and check if goldens are still failing

**Comment by rami-a** [comment]
There are a couple goldens that will have to be updated I think, like https://github.com/flutter/gallery/blob/master/golden_test/goldens/home_page_desktop_dark.png

You can see that the Material section has banner at the top, but your changes put appbar at the top, so I think you'll need to update the goldens

**Comment by rami-a** [comment]
When I ran `flutter test --update-goldens golden_test` on your branch, it showed different goldens than what you've included in this change. Are you running on a mac? If not then it wont work correctly.

File: android/app/src/main/res/drawable-v21/launch_background.xml [hunk 11]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (rami-a) [inline android/app/src/main/res/drawable-v21/launch_background.xml:1]: Could these not be included with this PR, they seem unrelated

File: android/app/src/main/res/values-night/styles.xml [hunk 12]
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (rami-a) [inline android/app/src/main/res/values-night/styles.xml:1]: Could these not be included with this PR, they seem unrelated

File: android/app/src/main/res/drawable-v21/launch_background.xml (hunk 11 above)
Comment (Alabhya268) [reply android/app/src/main/res/drawable-v21/launch_background.xml:1]: I didn't change them, It changed itself

**Comment by Alabhya268** [comment]
I've put back original ss for now as new ss res was different which caused test to fail .

**Comment by Alabhya268** [comment]
> When I ran `flutter test --update-goldens golden_test` on your branch, it showed different goldens than what you've included in this change. Are you running on a mac? If not then it wont work correctly.

I am running it on windows and yes golden test is still failing .

**Comment by Alabhya268** [comment]
> flutter test --update-goldens golden_test

I am running it on windows ,and golden test is failing for different reasons .

image

**Comment by rami-a** [comment]
You can see in the docs here, https://github.com/flutter/gallery/tree/master/golden_test#updating-goldens

that the goldens must be generated on mac in order for the check to pass. I've attached the updated goldens to this comment. You should be able to copy them into your branch and push them up, then it should pass hopefully!


home_page_desktop_light
home_page_desktop_dark

**Comment by Alabhya268** [comment]
> 
> 
> You can see in the docs here, https://github.com/flutter/gallery/tree/master/golden_test#updating-goldens
> 
> that the goldens must be generated on mac in order for the check to pass. I've attached the updated goldens to this comment. You should be able to copy them into your branch and push them up, then it should pass hopefully!
> 
> home_page_desktop_light
> home_page_desktop_dark

@rami ,very thanks for new screenshots.
Now all the tests are passing successfully .

**Comment by rami-a** [comment]
Thanks again @Alabhya268 !!
